src/main/java/com/iexec/worker/executor/TaskNotificationService.java

```diff
@@ -251,13 +250,25 @@ private TaskNotificationType updateStatusAndGetNextAction(String chainTaskId,
         return updateStatusAndGetNextAction(chainTaskId, statusUpdate);
     }
 
-    private TaskNotificationType updateStatusAndGetNextAction(String chainTaskId, ReplicateStatusUpdate statusUpdate) {
+    TaskNotificationType updateStatusAndGetNextAction(String chainTaskId, ReplicateStatusUpdate statusUpdate) {
         log.info("update replicate request [chainTaskId:{}, status:{}, details:{}]",
                 chainTaskId, statusUpdate.getStatus(), statusUpdate.getDetailsWithoutLogs());
 
         TaskNotificationType next = null;
+
         // As long as the Core doesn't reply, we try to contact it. It may be rebooting.
         while (next == null && !isFinalDeadlineReached(chainTaskId, Instant.now().toEpochMilli())) {
+            // Let's wait for the STOMP session to be ready.
+            // Otherwise, an update could be lost.
+            try {
+                subscriptionService.waitForSessionReady();
```

Comment:
Should we pass the task id as an argument and check the subscription to the topic has been properly established as well ?

Reply:
This is a good idea, but I'm not sure we currently know how to do so. This may need some work to be achieved. I don't think I would like to do it in this PR.

Reply:
I thought about adding sessionId (not taskId) to `Session[Created/Lost]Event` to avoid deadlocks, not sure if this could be related to your suggestion

Reply:
The current implementation is good as it is.
We will see later if it needs refinement.
(I was kind of worried of what could happen if a STOMP session is connected but the topic of the task has not been resubscribed, but we come to a corner case here)